Terminal.Gui/View/Layout/Dim.cs

```diff
@@ -0,0 +1,766 @@
+        var subviewsSize = 0;
+
+        int autoMin = MinimumContentDim?.Anchor (superviewContentSize) ?? 0;
+
+        if (superviewContentSize < autoMin)
+        {
+            Debug.WriteLine ($"WARNING: DimAuto specifies a min size ({autoMin}), but the SuperView's bounds are smaller ({superviewContentSize}).");
+
+            //return superviewContentSize;
+        }
+
+        if (Style.HasFlag (DimAutoStyle.Text))
+        {
+            textSize = int.Max (autoMin, dimension == Dimension.Width ? us.TextFormatter.Size.Width : us.TextFormatter.Size.Height);
+        }
+
+        if (Style.HasFlag (DimAutoStyle.Content))
+        {
+            if (us._contentSize is { })
+            {
+                subviewsSize = dimension == Dimension.Width ? us.ContentSize.Width : us.ContentSize.Height;
+            }
+            else
+            {
+                // TODO: AnchorEnd needs work
+                // TODO: If _min > 0 we can SetRelativeLayout for the subviews?
+                subviewsSize = 0;
+
+                if (us.Subviews.Count > 0)
+                {
+                    for (var i = 0; i < us.Subviews.Count; i++)
+                    {
+                        View v = us.Subviews [i];
+                        bool isNotPosAnchorEnd = dimension == Dimension.Width ? v.X is not PosAnchorEnd : v.Y is not PosAnchorEnd;
+
+                        //if (!isNotPosAnchorEnd)
+                        //{
+                        //    v.SetRelativeLayout(dimension == Dimension.Width ? (new Size (autoMin, 0)) : new Size (0, autoMin));
+                        //}
+
+                        if (isNotPosAnchorEnd)
+                        {
+                            int size = dimension == Dimension.Width ? v.Frame.X + v.Frame.Width : v.Frame.Y + v.Frame.Height;
+
+                            if (size > subviewsSize)
+                            {
+                                subviewsSize = size;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // All sizes here are content-relative; ignoring adornments.
+        // We take the larger of text and content.
+        int max = int.Max (textSize, subviewsSize);
+
+        // And, if min: is set, it wins if larger
+        max = int.Max (max, autoMin);
+
+        // Factor in adornments
+        Thickness thickness = us.GetAdornmentsThickness ();
+
+        if (dimension == Dimension.Width)
+        {
+            max += thickness.Horizontal;
+        }
+        else
+        {
+            max += thickness.Vertical;
+        }
+
+        // If max: is set, clamp the return - BUGBUG: Not tested
+        return int.Min (max, MaximumContentDim?.Anchor (superviewContentSize) ?? max);
+    }
+
+    internal override bool ReferencesOtherViews ()
+    {
+        // BUGBUG: This is not correct. _contentSize may be null.
+        return false; //_style.HasFlag (DimAutoStyle.Content);
```

Comment:
Oh, this reminds me of something unrelated.

If there is a relevant concept of "truth" for these types (like <= 0 or something being false and everything else true), `operator true` and `operator false` can be defined.

These types are a really borderline case for it, though, so this is VERY optional.